### ISSUE TITLE
pip installation broken

### ISSUE THREAD

**Issue by GuessWhoSamFoo** [MattDMo/NYTimesArticleAPI#2]
I'm not sure if this API wrapper is still being actively maintained.

Running pip install NYTimesArticleAPI results in this error.

```
Collecting NYTimesArticleAPI
  Could not find a version that satisfies the requirement NYTimesArticleAPI (from versions: )
No matching distribution found for NYTimesArticleAPI
```

EDIT:
Temporary workaround by installing via pip locally
```
pip install -e <Path to NYTimesArticleAPI>
```

**Comment by prateeky2806** [comment]
The pip install is not working and while installing pip locally I am getting an error
 
> "python setup.py egg_info" failed with error code 1 in /tmp/pip-ADB9WL-build/

Can anyone please fix this or suggest me some way to install this package.

**Comment by GuessWhoSamFoo** [comment]
Did you try updating `setuptools` yet?

**Comment by prateeky2806** [comment]
@GuessWhoSamFoo  
Yes, I have tried all the thing mentioned on this [page](http://stackoverflow.com/questions/35991403/python-pip-install-gives-command-python-setup-py-egg-info-failed-with-error-c) including upgrading setuptools. Can it be some problem with C compiler? I am using bash on windows but also tried it on ubuntu and it didn't work for me.

**Comment by GuessWhoSamFoo** [comment]
@prateeky2806 Hmm... I doubt it is the C compiler. Maybe try running `sudo pip` , that works sometimes. Other than that, I would check the logs to see if any dependencies are missing.

**Comment by nkramer44** [comment]
Did anyone ever figure out why pip install isn't working? The package shows up in pip search, but can't be installed for the same reason as @prateeky2806 was having.  setuptools and requests are both updated

**Comment by GuessWhoSamFoo** [comment]
Not really sure. I just did an install on a new virtual environment using Anaconda on Ubuntu 16.04 without any issues.

**Comment by ChampeBarton** [comment]
Has anybody figured this one out yet? Setuptools and requests are both updated and I'm getting 

`Could not find a version that satisfies the requirement nytimesarticleapi (from versions: )
No matching distribution found for nytimesarticleapi`

The package shows up in a pip install search, and shows up online @ https://pypi.python.org/pypi/nytimesarticleapi, too; however, that link says the package was authored by Brendan Ryan and his github homepage gives a 404error. Could really use some help!

I also notice in Matt's Progress Tracking he has yet to check off "upload to PyPI" ... could that be the problem?

**Comment by people-can-fly** [comment]
Basically i copied the code from `NYTimesArticleAPI/NYTimesArticleAPI/search_api.py` and replaced it with my installed `nytimesarticle` file `nytimesarticle.py`

Thus it removed 

`def _utf8_encode(self, d):
......`

which prevented `nytimesarticle` module to work with python3, throwing `TypeError:must be str,not bytes` on search function of the api.

**Comment by cbjrobertson** [comment]
Has anyone solved this?

Update: I replaced the `nytimesarticle` file `nytimesarticle.py` with `NYTimesArticleAPI/NYTimesArticleAPI/search_api.py` and it seems to be working fine, with no `Type Error: must be str, not bytes`. I don't know if this is a good solution, but it's working for me.

**Comment by ChampeBarton** [comment]
Not positive if this problem has been solved, but I ended up forking the whole package then essentially remaking the package with different identifying materials and loading it as a separate package onto my local drive. Check my github under my thesis source code to see if that provides any help.

Best,
Champe

**Comment by deveArt** [comment]
This 
`ModuleNotFoundError: No module named 'ez_setup'`

caused an installation error: 
`pip._internal.exceptions.InstallationError: Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-s52oqssv/NYTimesArticleAPInew/`

installation of ez_setup doesn't help. I think it should be removed from setup.py

**Comment by lud0** [comment]
@deveArt the same for me, I've removed it in my fork and now it's working.
